### PR TITLE
Add reg service URL to notification context

### PR DESCRIPTION
**Pull request by rajivnathan**
Fixes an issue where the registration service URL is not being set in the notification context.

### CODE REVIEW

File: controllers/masteruserrecord/sync.go [hunk 1]
```diff
@@ -195,9 +196,18 @@ func (s *Synchronizer) alignReadiness() (bool, error) {
 		// if there is no existing notification with these labels
 		if len(notificationList.Items) == 0 {
 
+			config, err := toolchainconfig.GetToolchainConfig(s.hostClient)
+			if err != nil {
+				return false, errs.Wrapf(err, "unable to get ToolchainConfig")
+			}
+
+			keysAndVals := map[string]string{
+				"RegistrationURL": config.RegistrationService().RegistrationServiceURL(),
```
Comment (alexeykazakov) [inline controllers/masteruserrecord/sync.go:205]: Let's use constants for notification keys.

File: controllers/usersignup/usersignup_controller.go [hunk 2]
```diff
@@ -557,12 +557,17 @@ func (r *Reconciler) DeleteMasterUserRecord(mur *toolchainv1alpha1.MasterUserRec
 	return nil
 }
 
-func (r *Reconciler) sendDeactivatingNotification(logger logr.Logger, userSignup *toolchainv1alpha1.UserSignup) error {
+func (r *Reconciler) sendDeactivatingNotification(logger logr.Logger, config toolchainconfig.ToolchainConfig, userSignup *toolchainv1alpha1.UserSignup) error {
+	keysAndVals := map[string]string{
+		"RegistrationURL": config.RegistrationService().RegistrationServiceURL(),
```
Comment (alexeykazakov) [inline controllers/usersignup/usersignup_controller.go:562]: And here.

File: controllers/usersignup/usersignup_controller.go [hunk 3]
```diff
@@ -574,12 +579,17 @@ func (r *Reconciler) sendDeactivatingNotification(logger logr.Logger, userSignup
 	return nil
 }
 
-func (r *Reconciler) sendDeactivatedNotification(logger logr.Logger, userSignup *toolchainv1alpha1.UserSignup) error {
+func (r *Reconciler) sendDeactivatedNotification(logger logr.Logger, config toolchainconfig.ToolchainConfig, userSignup *toolchainv1alpha1.UserSignup) error {
+	keysAndVals := map[string]string{
+		"RegistrationURL": config.RegistrationService().RegistrationServiceURL(),
```
Comment (alexeykazakov) [inline controllers/usersignup/usersignup_controller.go:584]: And here.

**Review by alexeykazakov — APPROVED** [review]
Looks good. But let's use some constant instead of hardcoding the same key multiple times. Also, shouldn't we also modify our e2e tests?

File: controllers/masteruserrecord/sync.go (hunk 1 above)
Comment (rajivnathan) [reply controllers/masteruserrecord/sync.go]: Done 39b6f7c058b917f01984d2f7b3268aa562d68e40

File: controllers/usersignup/usersignup_controller.go (hunk 2 above)
Comment (rajivnathan) [reply controllers/usersignup/usersignup_controller.go]: Done 39b6f7c058b917f01984d2f7b3268aa562d68e40

File: controllers/usersignup/usersignup_controller.go (hunk 3 above)
Comment (rajivnathan) [reply controllers/usersignup/usersignup_controller.go]: Done 39b6f7c058b917f01984d2f7b3268aa562d68e40

**Review by xcoulon — APPROVED** [review]
Looks good, but I assume we should also prepare a PR to update the ToolchainConfig with the Registration URL for prod and stage?

**Comment by rajivnathan** [comment]
@xcoulon the reg service URL is already configured in ToolchainConfig for both stage and prod, just reconfirmed.

**Comment by xcoulon** [comment]
> @xcoulon the reg service URL is already configured in ToolchainConfig for both stage and prod, just reconfirmed.

oh, my apologies, I missed that. I thought this was a new config parameter 🤦‍♂️

**Comment by rajivnathan** [comment]
@alexeykazakov we don't set the RegistrationServiceURL in e2e configuration so I guess unit tests should be good enough?

**Pull request merged**